transactions.go

```diff
@@ -96,3 +96,18 @@ func (t EndBattleTransaction) BattleLog() string {
 	// TODO: include reason the battle ended
 	return "The battle has ended."
 }
+
+// Handles pre-turn status checks. (Paralysis, Sleeping, etc.)
+type StatusEffectCheckTransaction struct {
+	Target       target
+	StatusEffect StatusCondition
+	Success      bool
+}
+
+func (t StatusEffectCheckTransaction) BattleLog() string {
+	failure := ""
+	if !t.Success {
+		failure = " and is unable to move"
+	}
+	return fmt.Sprintf("%s is %s%s.", t.Target.Pokemon.GetName(), t.Target.Pokemon.StatusEffects&NONVOLATILE_STATUS_MASK, failure)
+}
```

Comment:
Think this would be better wrapped up as something like `ImmobilizeTransaction` where the cause is passed in as a `StatusEffect`. That way the same transaction can be used for moves/abilities that cause immobilization.

Reply:
Yeah, I like that better.